src/store/reducers/web3.js

```diff
@@ -0,0 +1,20 @@
+import * as actionTypes from '../actions/actionTypes';
+
+const initialState = {
+  web3: null,
+};
+
+const setWeb3 = (state, web3) => {
+  return { web3: web3 };
+};
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case actionTypes.SET_WEB3:
```

Comment:
I don't like put complex objects into the redux state, prefer to define them in a service (see what I did with graphql client). But maybe people are doing this so guess it's ok

Reply:
I don't know what people do, I just came up with the idea...
The thing is that the web3 is null at the beginning and only initialized with the provider the user choses with web3modal (metamask, portis etc)

What do?